lib/Service/ColumnService.php

```diff
@@ -259,36 +212,36 @@ public function create(
 			if (!$found) {
 				break;
 			}
-			$newTitle = $title . ' (' . $i . ')';
+			$newTitle = $columnDto->getTitle() . ' (' . $i . ')';
 			$i++;
 		}
 
 		$time = new DateTime();
 		$item = new Column();
 		$item->setTitle($newTitle);
 		$item->setTableId($table->getId());
-		$item->setType($type);
-		$item->setSubtype($subtype !== null ? $subtype: '');
-		$item->setMandatory($mandatory);
-		$item->setDescription($description ?? '');
-		$item->setTextDefault($textDefault);
-		$item->setTextAllowedPattern($textAllowedPattern);
-		$item->setTextMaxLength($textMaxLength);
-		$item->setNumberDefault($numberDefault);
-		$item->setNumberMin($numberMin);
-		$item->setNumberMax($numberMax);
-		$item->setNumberDecimals($numberDecimals);
-		$item->setNumberPrefix($numberPrefix !== null ? $numberPrefix: '');
-		$item->setNumberSuffix($numberSuffix !== null ? $numberSuffix: '');
+		$item->setType($columnDto->getType());
+		$item->setSubtype($columnDto->getSubtype() ?? '');
+		$item->setMandatory($columnDto->isMandatory() ?? false);
+		$item->setDescription($columnDto->getDescription() ?? '');
+		$item->setTextDefault($columnDto->getTextDefault());
+		$item->setTextAllowedPattern($columnDto->getTextAllowedPattern());
+		$item->setTextMaxLength($columnDto->getTextMaxLength());
+		$item->setNumberDefault($columnDto->getNumberDefault());
+		$item->setNumberMin($columnDto->getNumberMin());
+		$item->setNumberMax($columnDto->getNumberMax());
+		$item->setNumberDecimals($columnDto->getNumberDecimals());
+		$item->setNumberPrefix($columnDto->getNumberPrefix() ?? '');
+		$item->setNumberSuffix($columnDto->getNumberSuffix() ?? '');
 		$this->updateMetadata($item, $userId, true);
-		$item->setSelectionOptions($selectionOptions);
-		$item->setSelectionDefault($selectionDefault);
-		$item->setDatetimeDefault($datetimeDefault);
-		$item->setUsergroupDefault($usergroupDefault);
-		$item->setUsergroupMultipleItems($usergroupMultipleItems);
-		$item->setUsergroupSelectUsers($usergroupSelectUsers);
-		$item->setUsergroupSelectGroups($usergroupSelectGroups);
-		$item->setShowUserStatus($showUserStatus);
+		$item->setSelectionOptions($columnDto->getSelectionOptions());
+		$item->setSelectionDefault($columnDto->getSelectionDefault());
+		$item->setDatetimeDefault($columnDto->getDatetimeDefault());
+		$item->setUsergroupDefault($columnDto->getUsergroupDefault());
+		$item->setUsergroupMultipleItems($columnDto->getUsergroupMultipleItems());
+		$item->setUsergroupSelectUsers($columnDto->getUsergroupSelectUsers());
+		$item->setUsergroupSelectGroups($columnDto->getUsergroupSelectGroups());
+		$item->setShowUserStatus($columnDto->getShowUserStatus());
```

Comment:
Maybe we can also move this assignment logic to the Column class to have something like `$item->setColumnSchema($columnDto)`?

Reply:
Or rather have a `ColumnEntity::fromDto` method?